lib/models/ObjectMDLocation.js

```diff
@@ -54,7 +60,11 @@ class ObjectMDLocation {
             'cryptoScheme',
             'cipheredDataKey',
         ].forEach(attrName => {
-            this._data[attrName] = location[attrName];
+            if (location[attrName] !== undefined) {
```

Comment:
should the check be against `null` or `undefined` instead?  (or any `falsy` value)

Reply:
`undefined` is what we want here, we want to check if the location does not contain such attribute, where this test would return `true`. If it is `null` we would want to copy the `null` value and not delete it.

Reply:
I always used `hasOwnProperty` for these checks but I don't see that used anywhere here. It is a very similar check but it also handles if that property is itself undefined.
```js
> a
{ a: 5, b: undefined }
> a['b'] !== undefined
false
> a.hasOwnProperty('b')
true
```

Idk if this changes the logic, just figured I would mention it.

Reply:
Right, I could have used it here. In case it would come with an existing attribute with `undefined` value for some reason (bug in code somewhere else, whatever) I think it's good to clean it up though.